addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiAqaraSensorWeatherV1Handler.java

```diff
@@ -0,0 +1,66 @@
+package org.openhab.binding.mihome.handler;
+
+import static org.openhab.binding.mihome.XiaomiGatewayBindingConstants.*;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.Thing;
+
+import com.google.gson.JsonObject;
+
+/**
+ * Handles the Xiaomi Aqara temperature, humidity and pressure sensor
+ *
+ * @author Daniel Walters - Support for Xiaomi Aqara temperature, humidity and pressure sensor
+ */
+public class XiaomiAqaraSensorWeatherV1Handler extends XiaomiSensorBaseHandler {
+
+    private static final String HUMIDITY = "humidity";
+    private static final String TEMPERATURE = "temperature";
+    private static final String VOLTAGE = "voltage";
+    private static final String PRESSURE = "pressure";
+
+    public XiaomiAqaraSensorWeatherV1Handler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    void parseReport(JsonObject data) {
+        parseDefault(data);
+    }
+
+    @Override
+    void parseHeartbeat(JsonObject data) {
+        parseDefault(data);
+    }
+
+    @Override
+    void parseReadAck(JsonObject data) {
+        parseDefault(data);
+    }
+
+    @Override
+    void parseWriteAck(JsonObject data) {
+        parseDefault(data);
+    }
+
+    @Override
+    void parseDefault(JsonObject data) {
```

Comment:
Here is an extract from my comms code:

```24-07-2017 10:55:46:542 [Thread-1] DEBUG xiaomiapi.ReceiveThread - RecvThread: {"cmd":"report","model":"weather.v1","sid":"158d0001a35ae2","short_id":63522,"data":"{\"temperature\":\"2341\"}"} from /192.168.0.28:4321```

```24-07-2017 10:55:46:552 [Thread-1] DEBUG xiaomiapi.ReceiveThread - RecvThread: {"cmd":"report","model":"weather.v1","sid":"158d0001a35ae2","short_id":63522,"data":"{\"humidity\":\"5347\"}"} from /192.168.0.28:4321```

```24-07-2017 10:55:46:562 [Thread-1] DEBUG xiaomiapi.ReceiveThread - RecvThread: {"cmd":"report","model":"weather.v1","sid":"158d0001a35ae2","short_id":63522,"data":"{\"pressure\":\"99837\"}"} from /192.168.0.28:4321```

```24-07-2017 10:55:46:574 [Thread-1] DEBUG xiaomiapi.ReceiveThread - RecvThread: {"cmd":"heartbeat","model":"weather.v1","sid":"158d0001a35ae2","short_id":63522,"data":"{\"voltage\":3005,\"temperature\":\"2341\",\"humidity\":\"5347\",\"pressure\":\"99837\"}"} from /192.168.0.28:4321```

As you can see the heartbeat and the report values match.